### PR TITLE
[java] Update rule UseEqualsToCompareStrings

### PR DESCRIPTION
**Pull request by adangel**
Part of #2701

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to pmd/7.0.x:<br />
This changeset changes 0 violations,<br />
introduces 18745 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/32e6e763d831462ab492cd895f4e680a/diff1/index.html">Full report</a></td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 30924 violations,<br />
introduces 56592 new violations, 3 new errors and 0 new configuration errors,<br />
removes 138203 violations, 8 errors and 3 configuration errors.<br />
<a href="https://chunk.io/pmd/32e6e763d831462ab492cd895f4e680a/diff2/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/resources/category/java/errorprone.xml [hunk 1]
```diff
@@ -3404,16 +3404,18 @@ public class Main {
           class="net.sourceforge.pmd.lang.rule.XPathRule"
           externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_errorprone.html#useequalstocomparestrings">
         <description>
-Using '==' or '!=' to compare strings only works if intern version is used on both sides.
-Use the equals() method instead.
+Using '==' or '!=' to compare strings only works if the internalized string (`String#intern()`)
```
Comment (oowekyala) [inline pmd-java/src/main/resources/category/java/errorprone.xml:3407]:
```suggestion
Using '==' or '!=' to compare strings is only reliable if the interned string (`String#intern()`)
```

File: pmd-java/src/main/resources/category/java/errorprone.xml [hunk 2]
```diff
@@ -3404,16 +3404,18 @@ public class Main {
...
+Use the `equals()` method instead.
         </description>
         <priority>3</priority>
         <properties>
             <property name="xpath">
                 <value>
 <![CDATA[
-//EqualityExpression
-    [count(PrimaryExpression[pmd-java:typeIs('java.lang.String')]) = 2]
+//InfixExpression[@Operator='==' or @Operator='!=']
```
Comment (oowekyala) [inline pmd-java/src/main/resources/category/java/errorprone.xml:3417]:
```suggestion
//InfixExpression[@Operator = ('==', '!=')]
```

**Review by oowekyala — APPROVED** [review]
LGTM

File: pmd-java/src/main/resources/category/java/errorprone.xml (hunk 2 above)
Comment (adangel) [reply pmd-java/src/main/resources/category/java/errorprone.xml]: nice! :+1:

**Pull request merged**